db/dcrsqlite/emptydb_test.go

```diff
@@ -0,0 +1,268 @@
+		t.Logf("RetrieveLatestBlockSummary() unexpected error: %v", err)
+	}
+}
+
+func checkRetrieveSDiff(t *testing.T, db *DB, i int64) {
+	res, err := db.RetrieveSDiff(i)
+	if err == nil {
+		t.Logf("RetrieveSDiff() failed: %v", res)
+	}
+	errMsg := fmt.Sprintf("%v", err)
+	if errMsg != "sql: no rows in result set" {
+		t.Logf("RetrieveSDiff() unexpected error: %v", err)
+	}
+}
+
+// TestRetrieveSDiff
+func TestRetrieveSDiff(t *testing.T) {
+	db := InitEmptyDB(t)
+
+	checkRetrieveSDiff(t, db, 0)
+	checkRetrieveSDiff(t, db, 1)
+	checkRetrieveSDiff(t, db, 2)
+}
+
+func checkRetrieveBlockHash(t *testing.T, db *DB, i int64) {
+	res, err := db.RetrieveBlockHash(i)
+	if err == nil {
+		t.Logf("RetrieveBlockHash() failed: %v", res)
+	}
+	errMsg := fmt.Sprintf("%v", err)
+	if errMsg != "sql: no rows in result set" {
+		t.Logf("RetrieveBlockHash() unexpected error: %v", err)
+	}
+}
+
+// TestRetrieveBlockHash
+func TestRetrieveBlockHash(t *testing.T) {
+	db := InitEmptyDB(t)
+
+	checkRetrieveBlockHash(t, db, 0)
+	checkRetrieveBlockHash(t, db, 1)
+	checkRetrieveBlockHash(t, db, 2)
+}
+
+func RangeTests(t *testing.T) {
+	db := InitEmptyDB(t)
+	//
+	db.RetrieveBlockSummaryByTimeRange(0, 0, 0)
+
+	db.RetrieveSDiffRange(0, 0)
+
+	db.RetrievePoolValAndSizeRange(0, 0)
+
+	db.RetrievePoolInfoRange(0, 0)
+}
+
+// Cover
+func TestCoverage(t *testing.T) {
+	db := InitEmptyDB(t)
+	{
+		_, err := db.RetrieveBlockHeight("")
+		if err == nil {
+			t.Logf("RetrieveBlockHeight() failed: error expeced")
+		}
+	}
+	{
+		_, err := db.RetrievePoolInfoByHash("")
+		if err == nil {
+			t.Logf("RetrievePoolInfoByHash() failed: error expeced")
+		}
+	}
+	{
+		null, zero, err := db.RetrieveWinnersByHash("")
+		if err == nil {
+			t.Logf("RetrieveWinnersByHash() failed: error expeced")
+		}
+		if null != nil {
+			t.Logf("RetrieveWinnersByHash() failed: nil expeced")
+		}
+		if zero != 0 {
+			t.Logf("RetrieveWinnersByHash() failed: 0 expeced")
+		}
+	}
+	{
+		null, zero, err := db.RetrieveWinners(0)
+		if err == nil {
+			t.Logf("RetrieveWinners() failed: error expeced")
+		}
+		if null != nil {
+			t.Logf("RetrieveWinners() failed: nil expeced")
+		}
+		if zero != "" {
+			t.Logf("RetrieveWinners() failed: empty string expeced")
+		}
+	}
+	{
+		result, err := db.RetrievePoolInfo(0)
+		if err == nil {
+			t.Logf("RetrievePoolInfo() failed: error expeced")
+		}
+		if result == nil {
+			t.Logf("RetrievePoolInfo() failed: default result expeced")
+		}
+		if result.Height != 0 {
+			t.Logf("RetrievePoolInfo() failed: invalid default result")
+		}
+		if result.Size != 0 {
+			t.Logf("RetrievePoolInfo() failed: invalid default result")
+		}
+		if result.ValAvg != 0 {
+			t.Logf("RetrievePoolInfo() failed: invalid default result")
+		}
+		if result.Value != 0 {
+			t.Logf("RetrievePoolInfo() failed: invalid default result")
+		}
+		if len(result.Winners) != 0 {
+			t.Logf("RetrievePoolInfo() failed: invalid default result\n" +
+				testutil.ArrayToString("result.Winners", result.Winners))
```

Comment:
@chappjc `RetrievePoolInfo(0)` fails the test here returning array that contains one element (empty string)